packages/jest-resolve-dependencies/src/index.ts

```diff
@@ -51,19 +52,32 @@ class DependencyResolver {
       if (this._resolver.isCoreModule(dependency)) {
         return acc;
       }
+
       let resolvedDependency;
+      let resolvedMockDependency;
       try {
         resolvedDependency = this._resolver.resolveModule(
           file,
           dependency,
           options,
         );
       } catch (e) {
-        resolvedDependency = this._resolver.getMockModule(file, dependency);
+        resolvedMockDependency = this._resolver.getMockModule(file, dependency);
```

Comment:
I'm not entirely sure but I assumed so, as the case where an error is thrown there expects the name from the metadata to exist but not be in the filesystem.

I left a note for prosterity in this commit: 8137d382f0b9a404f8157a8c612e03fdc458fa1b

I could try to get to the bottom of this in this commit since it's pretty related, would that be a good idea? 

If you do, is there a way I could figure out if this code is exercised in the test suite?